src/content/docs/en/guides/configuring-astro.mdx

```diff
@@ -1,131 +1,163 @@
 import { defineConfig } from 'astro/config'
 
 export default defineConfig({
-  // your configuration options here...
-  // https://docs.astro.build/en/reference/configuration-reference/
+  site: 'https://www.example.com',
+  base: '/docs',
+  trailingSlash: 'always',
 })
 ```
 
-You can also provide type definitions manually to VSCode, using this JSDoc notation:
+### Add site metadata
 
-```js
-// astro.config.mjs
-export default /** @type {import('astro').AstroUserConfig} */ {
-  // your configuration options here...
-  // https://docs.astro.build/en/reference/configuration-reference/
-}
+Astro does not use its configuration file for common SEO or meta data, only for information required to build your project code and render it to HTML.
+
+Instead, this information is added to your page `<head>` in HTML `<link>` and `<meta>` tags, just as if you were writing plain HTML pages.
+
+One common pattern for Astro sites is to create a `<Head />` [`.astro` component](/en/basics/astro-components/) that can be added to a common [layout component](/en/basics/layouts/) so it can apply to all your pages.
+
+```astro title="src/components/MainLayout.astro"
+---
+import Head from './Head.astro';
+const { ...props } = Astro.props;
+---
+<html>
+  <head>
+    <meta charset="utf-8">
+    <Head />
+    <!-- -->
+  </head>
+  <body><!-- --></body>
+</html>
 ```
 
-## Referencing Relative Files
+Because `Head.astro` is just a regular Astro component, you can import files and receive props passed from other components, such as a specific page title.
 
-If you provide a relative path to `root` or the `--root` CLI flag, Astro will resolve it against the current working directory where you ran the `astro` CLI command.
+```astro title="src/components/Head.astro"
+---
+import Favicon from '../assets/Favicon.astro';
+import SomeOtherTags from './SomeOtherTags.astro';
+const { title = 'My Astro Website', ...props } = Astro.props;
+---
+<link rel="sitemap" href="/sitemap-index.xml">
+<title>{title}</title>
+<meta name="description" content="Welcome to my new Astro site!">
 
-```js
-// astro.config.mjs
-import { defineConfig } from 'astro/config'
+<!-- Web analytics -->
+<script data-goatcounter="https://my-account.goatcounter.com/count" async src="//gc.zgo.at/count.js"></script>
 
-export default defineConfig({
-  // Resolves to the "./foo" directory in your current working directory
-  root: 'foo'
-})
+<!-- Open Graph tags -->
+<meta property="og:title" content="My New Astro Website" />
+<meta property="og:type" content="website" />
+<meta property="og:url" content="http://www.example.com/" />
+<meta property="og:description" content="Welcome to my new Astro site!" />
+<meta property="og:image" content="https://www.example.com/_astro/seo-banner.BZD7kegZ.webp">
+<meta property="og:image:alt" content="">
+
+<SomeOtherTags />
+
+<Favicon />
 ```
 
-Astro will resolve all other relative file and directory strings as relative to the project root:
+### Reference relative Files
```

Comment:
Neither this nor the “customize output filenames” examples are “common new project tasks” I don’t think. The filenames one is maybe a recipe? It should be a niche requirement (and one I’d recommend against unless someone _has_ to do this for some weird reason in their deploy setup). Definitely not something to recommend doing in a new project (and we’re mostly re-documenting Vite here, but necessarily so given it’s not so uncommon for someone to ask how to do this).

Referencing relative files also is _kind_ of niche, although admittedly very directly configuration-related. If the main usage is `root`, I might even be tempted to move this into the config reference for `root`…